packages/vite/src/server.ts

```diff
@@ -56,6 +57,7 @@ export async function buildServer (ctx: ViteBuildContext) {
       noDiscovery: true,
     },
     resolve: {
+      conditions: ((ctx.nuxt as any)._nitro as Nitro)?.options.exportConditions,
```

Comment:
Have you tried to externalize them instead? (so first step doesn't resolve deps)